tests/diff/float_values_test.go

```diff
@@ -76,6 +67,47 @@ func TestFloatValues(t *testing.T) {
 		}
 	})
 
+	t.Run("InsertNegativeZero", func(t *testing.T) {
+		t.Parallel()
+
+		filter := bson.D{{"_id", "1"}}
+		doc := bson.D{{"_id", "1"}, {"foo", math.Copysign(0.0, -1)}}
+
+		collection := db.Collection("insert-negative-zero")
+		_, err := collection.InsertOne(ctx, doc)
+		require.NoError(t, err)
+
+		var res bson.D
+		err = collection.FindOne(ctx, filter).Decode(&res)
+		require.NoError(t, err)
+
+		var actual float64
+		for _, e := range res {
+			if e.Key == "foo" {
+				var ok bool
+				actual, ok = e.Value.(float64)
+				require.True(t, ok)
+			}
+		}
+
+		expected := math.Copysign(0.0, +1)
+
+		// testify require equates -0 == 0 so this passes for both -0 and 0.
+		require.Equal(t, expected, actual)
+
+		t.Run("FerretDB", func(t *testing.T) {
+			t.Parallel()
+
+			require.Equal(t, math.Signbit(expected), math.Signbit(actual))
+		})
+
+		t.Run("MongoDB", func(t *testing.T) {
+			t.Parallel()
+
+			require.NotEqual(t, math.Signbit(expected), math.Signbit(actual))
```

Comment:
I believe you chose `require.NotEqual` over `require.Equal` on purpose, but I don't really understand why... Doesn't `require.Equal` look more readable? 

(Not a request for change, let's see what others think)

Reply:
You are right, I think it's easier to read and actually assert the equal value instead of checking notEqual. I've changed to below

```
		t.Run("FerretDB", func(t *testing.T) {
			t.Parallel()

			require.Equal(t, math.Signbit(math.Copysign(0.0, +1)), math.Signbit(actual))
		})

		t.Run("MongoDB", func(t *testing.T) {
			t.Parallel()

			require.Equal(t, math.Signbit(math.Copysign(0.0, -1)), math.Signbit(actual))
		})
```